torch/_refs/nn/functional/__init__.py

```diff
@@ -59,6 +61,58 @@
 
 Tensor = torch.Tensor
 
+
+# TODO: should we allow the user to set a different dtype for the mask generation?
+def _dropout_helper(
+    self: TensorLikeType,
+    val: float,
+) -> TensorLikeType:
+
+    return refs.lt(
+        refs.uniform(
+            self.shape, low=0.0, high=1.0, dtype=torch.float32, device=self.device
+        ),
+        val,
+    )
+
+
+@register_decomposition(torch.ops.aten.alpha_dropout)
+def alpha_dropout(
+    self: TensorLikeType, p: float = 0.5, training: bool = False, inplace: bool = False
+) -> TensorLikeType:
+
+    if inplace:
+        raise NotImplementedError
+
+    if not training:
+        return self
+
+    utils.check(
+        p <= 1 and p >= 0,
+        lambda: f"dropout probability has to be between 0 and 1, but got, {p}",
+    )
+
+    if p == 1:
+        return torch.zeros_like(self)
+
+    if p == 0:
+        return self
+
+    dropout_mask = _dropout_helper(self, 1 - p)
```

Comment:
I think this will always be a float32 tensor?

Reply:
I think this will be a tensor of boolean values. Please see:
```python
In [10]: input = torch.rand((2, 3), dtype=torch.float16)

In [11]: torch._refs.nn.functional._dropout_helper(input, val=0.8)
Out[11]: 
tensor([[ True,  True,  True],
        [ True,  True, False]])
```

Reply:
You're correct -- thank you!